docs/pages/code/operations/all.php

```diff
@@ -30,10 +26,13 @@
 };
 
 $collection = Collection::fromIterable($generator());
-print_r($collection->all()); // [0 => 'c', 1 => 'b', 2 => 'd']
+print_r($collection->all(false)); // [0 => 'c', 1 => 'b', 2 => 'd']
 
-// Example 2 -> standalone usage
-$even = static fn (int $value): bool => $value % 2 === 0;
+$collection = Collection::fromIterable($generator());
+print_r($collection->all()); // ['a', 'b', 'c', 'd']
+
+// Example 2 -> usage with "map" collection
+$collection = Collection::fromIterable(['foo' => 1, 'bar' => 2]);
 
-$piped = Pipe::of()(Filter::of()($even), All::of())(new ArrayIterator([1, 2, 3, 4]));
-print_r($piped); // [2, 4]
```

Comment:
this standalone usage is actually not valid. In order to use `Pipe` every operation needs to return an `Iterator`, but `All` returns an `array`.

This was missed when introducing the `All` operation because there isn't any enforcement of the return type at the [Operation](https://github.com/loophp/collection/blob/master/src/Contract/Operation.php) interface level. It's not easy to implement that however because you could return multiple closures and only the last one needs to return an `Iterator`.

I will look to make a separate PR after this to update the `All` operation to return a `Generator` though, and move the conversion to array back into the Collection method

Reply:
Indeed, let's fix that in the next PR.